diff --git a/build.zig b/build.zig
index f9d861b..363e0d1 100644
--- a/build.zig
+++ b/build.zig
@@ -136,18 +136,19 @@ pub fn build(b: *std.Build) !void {
         // As such, these build steps are lacking. For example, the Darwin
         // build only produces an xcframework.
 
-        // We shouldn't have this guard but we don't currently
-        // build on macOS this way ironically so we need to fix that.
+        libghostty_shared.installHeader();
+        libghostty_static.install("libghostty.a");
+
         if (!config.target.result.os.tag.isDarwin()) {
-            libghostty_shared.installHeader(); // Only need one header
             libghostty_shared.install("libghostty.so");
-            libghostty_static.install("libghostty.a");
         }
     }
 
     // macOS only artifacts. These will error if they're initialized for
     // other targets.
-    if (config.target.result.os.tag.isDarwin()) {
+    if (config.target.result.os.tag.isDarwin() and
+        (config.emit_xcframework or config.emit_macos_app))
+    {
         // Ghostty xcframework
         const xcframework = try buildpkg.GhosttyXCFramework.init(
             b,
@@ -204,7 +205,7 @@ pub fn build(b: *std.Build) !void {
 
         // On macOS we can run the macOS app. For "run" we always force
         // a native-only build so that we can run as quickly as possible.
-        if (config.target.result.os.tag.isDarwin()) {
+        if (config.target.result.os.tag.isDarwin() and config.emit_macos_app) {
             const xcframework_native = try buildpkg.GhosttyXCFramework.init(
                 b,
                 &deps,
